Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -456,12 +456,32 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
                     {
                         extractPath = _environment.WebRootPath;
                         FileSystemHelpers.EnsureDirectory(extractPath);
-
-                        tracer.Trace("Generate xdt file to map application to run from wwwroot when request for /{0}", package.Identity.Id);
-                        GenerateRemapToWebRootApplicationHostXdt(installationDirectory, string.Format(CultureInfo.InvariantCulture, "/{0}", package.Identity.Id));
                     }
 
                     await remoteRepo.DownloadPackageToFolder(package.Identity, extractPath, pathToLocalCopyOfNudpk: packageLocalFilePath);
+
+                    if (SiteExtensionInfo.SiteExtensionType.WebRoot == type)
+                    {
+                        // if install to WebRoot, check if there is any xdt file come with package
+                        // if there is one, move it to site extension folder
+                        // if not, generate one to map request to wwwroot
+                        string xdtFile = Path.Combine(extractPath, _applicationHostFile);
+                        if (File.Exists(xdtFile))
+                        {
+                            tracer.Trace("Use xdt file from package.");
+                            FileInfo xdtFileInfo = new FileInfo(xdtFile);
```

Comment:
But why do you need FileInfo at all here? Why not just `Path.Combine(installationDirectory, _applicationHostFile)`?